internal/controller/policy_subreconciler.go

```diff
@@ -0,0 +1,339 @@
+	if policy.IsMutating() {
+		if err := r.reconcileMutatingWebhookConfigurationDeletion(ctx, policy); err != nil {
+			return ctrl.Result{}, err
+		}
+	} else {
+		if err := r.reconcileValidatingWebhookConfigurationDeletion(ctx, policy); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+	// Remove the old finalizer used to ensure that the policy server created
+	// before this controller version is delete as well. As the upgrade path
+	// supported by the Kubewarden project does not allow jumping versions, we
+	// can safely remove this line of code after a few releases.
+	controllerutil.RemoveFinalizer(policy, constants.KubewardenFinalizerPre114)
+	controllerutil.RemoveFinalizer(policy, constants.KubewardenFinalizer)
+	if err := r.Update(ctx, policy); err != nil {
+		return ctrl.Result{}, fmt.Errorf("cannot update admission policy: %w", err)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *policySubReconciler) setPolicyStatus(ctx context.Context, policy policiesv1.Policy) error {
+	policyServerDeployment := appsv1.Deployment{}
+	policyServerDeploymentName := policyServerDeploymentName(policy.GetPolicyServer())
+
+	if err := r.Get(ctx, types.NamespacedName{Namespace: r.deploymentsNamespace, Name: policyServerDeploymentName}, &policyServerDeployment); err != nil {
+		if apierrors.IsNotFound(err) {
+			// If the policy server deployment is not found, the policy is not scheduled
+			return nil
+		}
+
+		return errors.Join(errors.New("could not get policy server deployment"), err)
+	}
+
+	policyServerConfigMap := corev1.ConfigMap{}
+	if err := r.Get(ctx, types.NamespacedName{Namespace: r.deploymentsNamespace, Name: policyServerDeploymentName}, &policyServerConfigMap); err != nil {
+		return errors.Join(errors.New("could not get configmap"), err)
+	}
+
+	policyMap, err := getPolicyMapFromConfigMap(&policyServerConfigMap)
+	if err == nil {
+		if policyConfig, ok := policyMap[policy.GetUniqueName()]; ok {
+			policy.SetPolicyModeStatus(policiesv1.PolicyModeStatus(policyConfig.PolicyMode))
+		} else {
+			policy.SetPolicyModeStatus(policiesv1.PolicyModeStatusUnknown)
+		}
+	} else {
+		policy.SetPolicyModeStatus(policiesv1.PolicyModeStatusUnknown)
+	}
+
+	policyStatus := policy.GetStatus()
+	setPolicyConfigurationCondition(&policyServerConfigMap, &policyServerDeployment, &policyStatus.Conditions)
+
+	return nil
+}
+
+func (r *policySubReconciler) getPolicyServer(ctx context.Context, policy policiesv1.Policy) (*policiesv1.PolicyServer, error) {
+	policyServer := policiesv1.PolicyServer{}
+	if err := r.Get(ctx, types.NamespacedName{Name: policy.GetPolicyServer()}, &policyServer); err != nil {
+		return nil, errors.Join(errors.New("could not get policy server"), err)
+	}
+	return &policyServer, nil
+}
+
+func (r *policySubReconciler) isPolicyUniquelyReachable(ctx context.Context, policyServerDeployment *appsv1.Deployment, policyName string) bool {
```

Comment:
Can you add some documentation explaining what this method is doing?

Reply:
from my understanding, this is some technical debt we will address in the near future. You "just" moved this method around.

I'm fine addressing all of that later on